src/store/login/actions.js

```diff
@@ -0,0 +1,62 @@
+import { createAction } from 'redux-actions';
+import { push } from 'connected-react-router';
+import {
+  USERNAME_INPUT_VALUE,
+  PASSWORD_INPUT_VALUE,
+  LOGIN_FAILURE,
+  LOGIN_SUCCESS,
+  EMPTY_FIELDS_ERROR
+} from './types';
+import { post } from '../../utils/http';
+import { LOGIN_ENDPOINT, LOGIN_URL } from '../../config';
+import { set as setIntoLocalStorage, get as getFromLocalStorage } from '../../utils/localStorage';
+import { setUserData } from '../user/actions';
+import { createHomeRoute } from '../../utils/routeCreators';
+
+export const changeUsernameValue = createAction(USERNAME_INPUT_VALUE, (value) => value);
+export const changePasswordValue = createAction(PASSWORD_INPUT_VALUE, (value) => value);
+export const loginSuccess = createAction(LOGIN_SUCCESS, (data) => data);
+export const loginFailure = createAction(LOGIN_FAILURE);
+export const emptyFieldsError = createAction(EMPTY_FIELDS_ERROR);
+
+export const login = () => (dispatch, getState) => {
+  const { login: { username, password } } = getState();
+
+  if (!username || !password) return dispatch(emptyFieldsError());
```

Comment:
NIP: In my opinion, it is not the best idea to write if conditions in one line
If you need to add more functionality in it, you will need to add brackets

Reply:
fixed